moto/glue/utils.py

```diff
@@ -20,6 +22,7 @@
     one_of,
     pyparsing_common,
 )
+from typing_extensions import Self
```

Comment:
Same issue here - `typing_extensions` is a transitive dependency, so this may not always exist. It would be better to use the explicit type, instead of `Self`.